CLOUDP-162772: Apply oneOf transformation globally for enums (#383)
mongodb/go-client-mongodb-atlas: tools/transformer/README.md

## Implemented transformations

1. oneOf transformation

> Appliet to all objects that meet either of the following criteria:

Reviewer: ```suggestion
Applied to all objects that meet either of the following criteria:
```

> Have extension "x-xgen-go-transform": "oneOfMerge"

> > Have "oneOf" field and all the objects referenced are enums

Reviewer: ```suggestion
> Have "oneOf" field with all referenced schemas having an enum type
```
